lib/emscripten/src/lib.rs

```diff
@@ -104,1116 +112,1235 @@ macro_rules! mock_external {
+
+        data.push((
+            "STACKTOP".into(),
+            LocalGlobal { data: stacktop(STATIC_BUMP) as _ },
+            GlobalDesc { mutable: false, ty: Type::I32 }),
+        );
+
+        data.push((
+            "DYNAMICTOP_PTR",
+            LocalGlobal { data: dynamictop_ptr(STATIC_BUMP) as _ },
+            GlobalDesc { mutable: false, ty: Type::I32 }),
+        );
+
+        data.push((
+            "Infinity",
+            LocalGlobal { data: std::f64::INFINITY.to_bits() },
+            GlobalDesc { mutable: false, ty: Type::F64 },
+        ));
+
+        data.push((
+            "NaN",
+            LocalGlobal { data: std::f64::NAN.to_bits() },
+            GlobalDesc { mutable: false, ty: Type::F64 },
+        ));
+
+        data.push((
+            "tableBase",
+            LocalGlobal { data: 0 },
+            GlobalDesc { mutable: false, ty: Type::I32 },
+        ));
+
+        Self {
+            data,
+        }
+    }
+}
+
+
 pub fn generate_emscripten_env() -> Imports {
-    let mut import_object = Imports::new();
+    use Type::*;
 
-    //    import_object.add(
-    //        "spectest".to_string(),
-    //        "print_i32".to_string(),
-    //        Import::Func(
-    //            print_i32 as _,
-    //            FuncSig {
-    //                params: vec![Type::I32],
-    //                returns: vec![],
-    //            },
-    //        ),
-    //    );
-    //
-    //    import_object.add(
-    //        "spectest".to_string(),
-    //        "global_i32".to_string(),
-    //        Import::Global(Value::I64(GLOBAL_I32 as _)),
-    //    );
 
-    // Globals
-    import_object.add(
-        "env".to_string(),
-        "STACKTOP".to_string(),
-        Import::Global(Value::I64(stacktop(STATIC_BUMP) as _)),
-    );
-    import_object.add(
-        "env".to_string(),
-        "STACK_MAX".to_string(),
-        Import::Global(Value::I64(stack_max(STATIC_BUMP) as _)),
-    );
-    import_object.add(
-        "env".to_string(),
-        "DYNAMICTOP_PTR".to_string(),
-        Import::Global(Value::I64(dynamictop_ptr(STATIC_BUMP) as _)),
-    );
-    import_object.add(
-        "global".to_string(),
-        "Infinity".to_string(),
-        Import::Global(Value::I64(std::f64::INFINITY.to_bits() as _)),
-    );
-    import_object.add(
-        "global".to_string(),
-        "NaN".to_string(),
-        Import::Global(Value::I64(std::f64::NAN.to_bits() as _)),
-    );
-    import_object.add(
-        "env".to_string(),
-        "tableBase".to_string(),
-        Import::Global(Value::I64(0)),
-    );
-    //    // Print functions
+    let mut imports = Imports::new();
+    let mut env_imports= HashMap::new();
+    let em_globals = EmscriptenGlobals::new();
+
+    // Add globals.
+    for (name, ref global, desc) in em_globals.data {
+        let export = Export::Global {
+            local: global as *mut _,
+            global: desc,
+        };
+
+        imports.register_export("env", name, export);
+    }
```

Comment:
We can add globals directly, as we are doing with the other imported functions.
No need to depend in a EmscriptenGlobals abstraction (neither to walk an array at runtime).

Reply:
This impl is not complete, but this is needed because unlike previous Import impl, `Export::Global` hold pointer to the global to instead of moving the global value.

This means we can't create the globals before `generate_emscripten_env` and pass them in. In this case I'm using EmdcriptenGlobals

Reply:
Fixed the issue with previous impl.

Reply:
Can we hold the pointers to static variables instead?